### ISSUE TITLE
Need ember-template-compiler in test

### ISSUE THREAD

**Issue by jamesarosen** [ember-cli/ember-cli#3497]
Many component tests need the template compiler at runtime. Thus, I think it makes sense to import that library in test-mode only.

Suggested change:

```
// blueprints/app/files/Brocfile.js:

var app = new EmberApp();

app.import({
  test: 'bower_components/ember/ember-template-compiler.js'
});
```

Alternatively, we could have ember-testing or another addon automatically add `ember-template-compiler.js` in test mode.

**Comment by rwjblue** [comment]
I'd rather setup easy use of ES6 template strings and precompile in node-land (via something like https://github.com/rwjblue/broccoli-ember-inline-template-compiler).

Main reasoning:  

In a production environment the template compiler will most likely not be present, and if we always add the template compiler when running tests it is likely that you would never know that your application is completely broken in production (by way of the tests that is).

**Comment by rwjblue** [comment]
Also, to be clear, we absolutely should solve this problem.  I would just rather do the precompilation server side...

**Comment by jamesarosen** [comment]
I think the defensiveness is totally warranted. I wouldn't want to be surprised when deploying.

**Comment by rwjblue** [comment]
> I wouldn't want to be surprised when deploying.

Sadly, this is exactly what happened to me.  I had an addon that was using `Ember.Handlebars.compile` at runtime and didn't realize it until I did a production deploy. :crying_cat_face:

**Comment by jamesarosen** [comment]
@rwjblue's https://github.com/rwjblue/broccoli-ember-inline-template-compiler may be a really good alternative here.

**Comment by jamesarosen** [comment]
https://github.com/pangratz/ember-cli-htmlbars-inline-precompile may also be of use here

**Comment by stefanpenner** [comment]
@jamesarosen yes that is the plan :)

**Comment by rwjblue** [comment]
Yes. There are some related PR's to be landed in that repo and ember-cli-babel, but once they land (hint hint @stefanpenner ) we plan to ship this setup by default.

**Comment by stefanpenner** [comment]
no longer needed, we use babel to allow inline HBS transpilation.
